Update kubernetes cluster autoscaler docs to add annotations (#17725)
DataDog/integrations-core: kubernetes_cluster_autoscaler/README.md

2. [Restart the Agent][5].

#### Metric collection

> Make sure that the Prometheus-formatted metrics are exposed in your kubernetes_cluster_autoscaler cluster. 
> For the Agent to start collecting metrics, the kubernetes_cluster_autoscaler pods need to be annotated.

Reviewer: ```suggestion
Make sure that the Prometheus-formatted metrics are exposed in your `kubernetes_cluster_autoscaler` cluster. 
For the Agent to start collecting metrics, the `kubernetes_cluster_autoscaler` pods need to be annotated.
```

> Kubernetes Cluster Autoscaler has metrics and livenessProbe endpoints that can be accessed on port 8085 (you can change that using the --address flag). These endpoints are located under `/metrics` and `/health-check` and provide valuable information about the state of your cluster during scaling operations.

Reviewer 2: non blocking nit. Is it possible to include the links to the cluster autoscaler docs where these are documented?

> Kubernetes Cluster Autoscaler has metrics and livenessProbe endpoints that can be accessed on port 8085 (you can change that using the --address flag). These endpoints are located under `/metrics` and `/health-check` and provide valuable information about the state of your cluster during scaling operations.

Reviewer: ```suggestion
Kubernetes Cluster Autoscaler has metrics and livenessProbe endpoints that can be accessed on port `8085`. These endpoints are located under `/metrics` and `/health-check` and provide valuable information about the state of your cluster during scaling operations.

**Note**: To change the default port, use the `--address` flag.
```

> To get the Cluster Autoscaler to expose some metrics for you. It's pretty simple, you have to: 

Reviewer: ```suggestion
To configure the Cluster Autoscaler to expose metrics, do the following:
```

> a) enable access to the `/metrics` route, expose port 8085 for your Cluster Autoscaler deployment:

Reviewer: ```suggestion
1. Enable access to the `/metrics` route and expose port `8085` for your Cluster Autoscaler deployment:
```

```
ports:
--name: app
containerPort: 8085
``` 

b) instruct your Prometheus to scrape it, by adding the following annotation to your Cluster Autoscaler service:
```
prometheus.io/scrape: true
```

**Note**: The listed metrics can only be collected if they are available. Some metrics are generated only when certain actions are performed. 

The only parameter required for configuring the kubernetes_cluster_autoscaler check is:

> - `openmetrics_endpoint`: This parameter should be set to the location where the Prometheus-formatted metrics are exposed. The default port is `8085`, but it can be configured using the `METRICS_PORT` [environment variable][10]. In containerized environments, `%%host%%` should be used for [host autodetection][3]. 

Reviewer 2: Link 10 is missing

> The only parameter required for configuring the kubernetes_cluster_autoscaler check is:
> - `openmetrics_endpoint`: This parameter should be set to the location where the Prometheus-formatted metrics are exposed. The default port is `8085`, but it can be configured using the `METRICS_PORT` [environment variable][10]. In containerized environments, `%%host%%` should be used for [host autodetection][3]. 

Reviewer: ```suggestion
The only parameter required for configuring the `kubernetes_cluster_autoscaler` check is `openmetrics_endpoint`. This parameter should be set to the location where the Prometheus-formatted metrics are exposed. The default port is `8085`. To configure a different port, use the `METRICS_PORT` [environment variable][10]. In containerized environments, `%%host%%` should be used for [host autodetection][3]. 
```
